### DISCUSSION TITLE
How to create a mass mutatation in .net 4.6

### DISCUSSION THREAD

**Discussion by robfolks** [graphql-dotnet/graphql-client#584] [Q&A]
Hi,

Is it possible to create a mutation and pass a list of objects to that mutation in .net 4.6? Currently I create a GraphQlRequest and write the query and add variables. This works for a single mutation but I would like the request to contain multiple objects in a single request.

```csharp
var request = new GraphQLRequest
                {
                    Query = @"mutation ($transactionSource: String, $trRegNumber: Int, $postedDate: String, $account: String, $glObject1: String, $subledger: String, $invoiceDate: String, $invoiceNumber: String, $dueDate: String, $text: String, $invoiceAmount: MoneyString, $amount: MoneyString, $jobLevel: Int, $workflow: Boolean, $ownerCode: Int)
                              {
                                addGLImportItems(inputs:[{node:{
                                transactionSource: {code: $transactionSource}
                                trRegNumber: $trRegNumber
                                postedDate: $postedDate
                                account: { code: $account}
                                glObject1: {code: $glObject1}
                                subledger: {code: $subledger}
                                invoiceDate: $invoiceDate
                                invoiceNumber: $invoiceNumber
                                dueDate: $dueDate
                                text: $text
                                invoiceAmount: $invoiceAmount
                                amount: $amount
                                jobLevel: { dbSubId: $jobLevel}
                                workflow: $workflow
                                owner: {ownerCode: $ownerCode }
                              }
                              }]) {edges { node { dbId }}}}",

                    Variables = new { input.TransactionSource, input.TrRegNumber, input.PostedDate, input.Account, input.GlObject1, input.Subledger, input.InvoiceDate, input.InvoiceNumber, input.DueDate, input.Text, input.InvoiceAmount, input.Amount, input.JobLevel, input.Workflow ,input.OwnerCode },
}
```

**Comment by rose-a** [comment]
Well... if you control the server implementation, then you can certainly create a mutation which accepts an array of input objects...

But a you cannot input multiple objects into a mutation which expects a single input object for "batch processing", you need to call it separately for each mutation.

**Reply by robfolks** [reply]
Hi, 

Unfortuantely I'm just a consumer of the API but the mutation can acccept multiple nodes in the inputs array. I perhaps didn't word my initial question very well. I'm struggling to find an example of how I could pass an array of nodes in a request.

**Reply by rose-a** [reply]
1. Create a class for your `inputs` type. As far as I can tell this expects a list of objects containing a `node` field which then contains the other fields.
2. Change your query string to accept a single variable of type `[InputType]` (or whatever the expected type of the `inputs` parameter of `addGLImportItems` is)
3. put your list into your variables object and make sure it is correctly named
